server/app/core/migrator.py

```
@@ -0,0 +1,195 @@
+                )
+            continue  # already applied
+        pending.append((mig_id, data.decode("utf-8"), cs))
+
+    if not pending:
+        logging.getLogger(__name__).info(
+            "Migrations: no pending files for plugin '%s'", plugin_name
+        )
+        return
+
+    # apply all pending migrations in a single transaction
+    assert _db.engine is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)